lib/manageiq/api/client/client.rb

```diff
@@ -31,6 +30,25 @@ def load_definitions
         @collections   = load_collections(entrypoint["collections"])
       end
 
+      def update_authentication(auth_options = {})
+        return @authentication unless ManageIQ::API::Client::Authentication.options?(auth_options)
+        saved_auth = @authentication
+        @authentication = ManageIQ::API::Client::Authentication.new(auth_options)
+        begin
+          reconnect
+        rescue => err
+          @authentication = saved_auth
+          raise err
+        end
+        @authentication
+      end
+
+      def reconnect
+        new_connection = ManageIQ::API::Client::Connection.new(self, options.slice(:ssl))
+        @connection = new_connection
```

Comment:
What I mean is...if the right hand side blows up, it should not change the `@connection` variable

```ruby
[1] pry(main)> @variable = 123
=> 123
[2] pry(main)> @variable = 1 / 0
ZeroDivisionError: divided by 0
from (pry):2:in `/'
[3] pry(main)> @variable
=> 123
```